libgalois/include/katana/Properties.h

```diff
@@ -385,23 +475,33 @@ struct LargeStringReadOnlyProperty {
   using ViewType = StringPropertyReadOnlyView<arrow::LargeStringArray>;
 };
 
-template <typename Props>
-Result<std::shared_ptr<arrow::Table>>
-AllocateTable(uint64_t num_rows, const std::vector<std::string>& names) {
-  constexpr auto num_tuple_elem = std::tuple_size<Props>::value;
-  static_assert(num_tuple_elem != 0);
-  std::shared_ptr<arrow::Table> table;
-  std::vector<katana::PropertyArrowTuple<Props>> rows(num_rows);
-  KATANA_LOG_ASSERT(names.size() == num_tuple_elem);
-  // TODO(gill): Replace this with NUMA allocated buffers.
-  if (auto r = arrow::stl::TableFromTupleRange(
-          arrow::default_memory_pool(), std::move(rows), names, &table);
-      !r.ok()) {
-    KATANA_LOG_DEBUG("arrow error: {}", r);
-    return Result<std::shared_ptr<arrow::Table>>(katana::ErrorCode::ArrowError);
+template <typename T>
+struct StructProperty {
+  using ArrowType = arrow::FixedSizeBinaryType;
+  using ViewType = katana::PODPropertyView<T>;
+
+  static katana::Result<std::shared_ptr<arrow::Table>> Allocate(
+      size_t num_rows, const std::string& name) {
+    auto res = arrow::FixedSizeBinaryType::Make(sizeof(T));
+    if (!res.ok()) {
+      return KATANA_ERROR(katana::ErrorCode::ArrowError, "{}", res.status());
```

Comment:
Including extra contextual information with the errors is useful. In this case it might be `"failed to make fixed size type: {}"` or something.

Please take a look at your other error returns.